lib/client.js

```diff
@@ -19,7 +19,7 @@ module.exports = class {
     constructor(engine, options) {
 
         Hoek.assert(engine, 'Missing catbox client engine');
-        Hoek.assert(typeof engine === 'object' || typeof engine === 'function', 'engine must be an engine object or engine prototype (function)');
+        Hoek.assert((typeof engine === 'object' && typeof engine.start === 'function') || typeof engine === 'function', 'engine must be an engine object or engine prototype (function)');
```

Comment:
This will produce an upfront error as opposed to some weird runtime error when interacting with the connection for the first time.

Not really sure how specific this validation should be for `engine objects`. Should I go and validate it implements all required methods or just one is good enough?

Reply:
I think just one is enough for the kind of gut-check that we're trying to do here.  If we wanted to check for a very catbox-specific method name we could use `validateSegmentName()`, but I also think `start()` does the trick.